src/averageAges.js

```diff
@@ -54,8 +82,36 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const onlySon = people.filter((person) =>
+    onlyWithSon
+      ? person.sex === 'm'
+        && people.some((mom) => {
+          if (person.mother === mom.name) {
+            person['motherBorn'] = mom.born;
+
+            return true;
+          }
+        })
+      : people.some((mom) => {
+        if (person.mother === mom.name) {
+          person['motherBorn'] = mom.born;
+
+          return true;
+        }
+      })
```

Comment:
It's really hard to read. Try to refactor it. Move a ternary operator to a some() method.  This is a redundant action `person['motherBorn'] = mom.born;`. You don't need to add an additional key to the person's object